### PR TITLE
Change Tools TFMs to net7.0

### PR DESCRIPTION
**Pull request by lbussell**
Roslyn analyzers needs a target framework of .NET 7.0 for source build to build without prebuilts. There are some projects in the Tools/ directory of roslyn-analyzers that still target net6.0.

### CODE REVIEW

**Comment by lbussell** [comment]
I've been trying to get this to work but keep running into the error:

```
CS8032: Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.
```

I'm not sure where we are finding that version of Microsoft.CodeAnalysis, I can't find any references to it in this repo.

**Comment by Youssef1313** [comment]
@lbussell I think this is the version used by dotnet/runtime. The compiler used in the repository is likely a version lower than 4.2.0.

**Review by mavasani — APPROVED** [review]
LGTM

For the build failure due to new IDExxxx code style diagnostics, feel free to set the diagnostic to an IDE suggestion in the .editorconfig file at the repo root and please file a separate tracking issue to fix the violations and revert the .editorconfig change.

For example: https://github.com/dotnet/roslyn-analyzers/blob/793113a41d7f21b03470521bf48438f2abd9b12f/.editorconfig#L243

**Comment by lbussell** [comment]
@mavasani, I filed an issue like you suggested: https://github.com/dotnet/roslyn-analyzers/issues/6128

I don't know what to do about the RS2003 errors:

`error RS2003: (NETCORE_ENGINEERING_TELEMETRY=Build) Rule 'RS0001' was shipped in analyzer release '3.3.0', but is no longer a supported diagnostic for any analyzer. Add an entry for this rule in a 'Removed Rules' table to unshipped file.`

I would try to edit the markdown files but they say not to edit them manually.

File: .editorconfig
Comment (Youssef1313) [inline .editorconfig:23]: Once https://github.com/dotnet/roslyn-analyzers/pull/6117 is merged, these should be removed.

File: eng/Versions.props
Comment (Youssef1313) [inline eng/Versions.props:42]: Once https://github.com/dotnet/roslyn-analyzers/pull/6117 is merged, it will produce a conflict here. The version from https://github.com/dotnet/roslyn-analyzers/pull/6117 should be used.

**Review by Youssef1313 — CHANGES_REQUESTED** [review]
Can this wait until https://github.com/dotnet/roslyn-analyzers/pull/6117 is merged?

**Comment by Youssef1313** [comment]
> I don't know what to do about the RS2003 errors:

They'll go away once https://github.com/dotnet/roslyn-analyzers/pull/6117 is merged.

**Comment by lbussell** [comment]
> Can this wait until #6117 is merged?

It can wait as long as we can get both changes into 7.0 RC2.

**Comment by mavasani** [comment]
#6117 has been merged

File: src/Tools/ReleaseNotesUtil/ReleaseNotesUtil.csproj
Comment (Youssef1313) [inline src/Tools/ReleaseNotesUtil/ReleaseNotesUtil.csproj:4]: Consider updating this line:

https://github.com/dotnet/roslyn-analyzers/blob/dc672ef14b6c6ce5db6e9c0ee39e47e07c9f8510/PostReleaseActivities.md#L28

**Review by Youssef1313 — APPROVED** [review]
LGTM with a minor documentation update.

I think it should be okay to update the documentation in a separate follow-up PR if you want to avoid waiting for CI again for this PR.

**Comment by lbussell** [comment]
CI was broken due to https://github.com/dotnet/arcade/issues/10248 anyways so I updated it.

**Comment by lbussell** [comment]
@mavasani it looks like all the checks passed. Or did some of them not run?

**Comment by mavasani** [comment]
> @mavasani it looks like all the checks passed. Or did some of them not run?

Yes we should be good to merge after an approval

**Review by mavasani — APPROVED** [review]

**Pull request merged**
